### PR TITLE
ignore unreadable metrics on certain closed systems

### LINKED ISSUE #14233: Opening /proc/[pid]/io: permission denied
I'm trying to run minio in distributed mode - 8 nodes with Prometheus monitoring. Everything works fine, got statistics on dashboard etc. except that every node got repeating error in log file:

```bash
API: SYSTEM()
Time: 16:39:32 CET 02/02/2022
DeploymentID: 
Error: open /proc/1074/io: permission denied (*fs.PathError)
       8: internal/logger/logonce.go:52:logger.(*logOnceType).logOnceIf()
       7: internal/logger/logonce.go:92:logger.LogOnceIf()
       6: cmd/metrics-v2.go:885:cmd.getMinioProcMetrics.func1()
       5: cmd/metrics-v2.go:211:cmd.cachedRead.func1.1()
       4: cmd/utils.go:887:cmd.(*timedValue).Get()
       3: cmd/metrics-v2.go:218:cmd.cachedRead()
       2: cmd/metrics-v2.go:1728:cmd.populateAndPublish()
       1: cmd/metrics-v2.go:1694:cmd.ReportMetrics.func1()
```

When I check permissions on /proc for running service:

```bash
● minio.service - MinIO
   Loaded: loaded (/etc/systemd/system/minio.service; enabled; vendor preset: disabled)
   Active: active (running) since Wed 2022-02-02 10:09:08 CET; 7h ago
     Docs: https://docs.min.io
  Process: 1068 ExecStartPre=/bin/bash -c if [ -z "${MINIO_VOLUMES}" ]; then echo "Variable MINIO_VOLUMES not set in /etc/default/minio"; exit 1; fi (code=exited, status=0/SUCCESS)
 Main PID: 1074 (minio)
    Tasks: 14
   Memory: 4.1G
   CGroup: /system.slice/minio.service
           └─1074 /usr/local/bin/minio server --address mds-1.local:9000 --console-address mds.local:443 --certs-dir /etc/ssl/minio https://mds-{1...8}.local/ds1/minio
```

Only few got owner as minio-user and **io** got permisison to be read only by root

```bash
dr-xr-xr-x.   9 minio-user minio-user 0 Feb  2 16:36 .
dr-xr-xr-x. 214 root       root       0 Feb  2 10:08 ..
-r--r--r--.   1 root       root       0 Feb  2 17:16 arch_status
dr-xr-xr-x.   2 minio-user minio-user 0 Feb  2 16:39 attr
-r--------.   1 root       root       0 Feb  2 17:16 auxv
-r--r--r--.   1 root       root       0 Feb  2 16:39 cgroup
--w-------.   1 root       root       0 Feb  2 17:16 clear_refs
-r--r--r--.   1 root       root       0 Feb  2 16:36 cmdline
-rw-r--r--.   1 root       root       0 Feb  2 16:39 comm
-rw-r--r--.   1 root       root       0 Feb  2 17:16 coredump_filter
-r--r--r--.   1 root       root       0 Feb  2 17:16 cpuset
lrwxrwxrwx.   1 root       root       0 Feb  2 17:16 cwd -> /usr/local
-r--------.   1 root       root       0 Feb  2 17:16 environ
lrwxrwxrwx.   1 root       root       0 Feb  2 16:39 exe -> /usr/local/bin/minio
dr-x------.   2 root       root       0 Feb  2 16:36 fd
dr-x------.   2 root       root       0 Feb  2 17:16 fdinfo
-rw-r--r--.   1 root       root       0 Feb  2 17:16 gid_map
-r--------.   1 root       root       0 Feb  2 16:36 io
-r--r--r--.   1 root       root       0 Feb  2 16:36 limits
-rw-r--r--.   1 root       root       0 Feb  2 16:39 loginuid
dr-x------.   2 root       root       0 Feb  2 17:16 map_files
-r--r--r--.   1 root       root       0 Feb  2 17:16 maps
-rw-------.   1 root       root       0 Feb  2 17:16 mem
-r--r--r--.   1 root       root       0 Feb  2 17:16 mountinfo
-r--r--r--.   1 root       root       0 Feb  2 17:16 mounts
-r--------.   1 root       root       0 Feb  2 17:16 mountstats
dr-xr-xr-x.  53 minio-user minio-user 0 Feb  2 17:16 net
dr-x--x--x.   2 root       root       0 Feb  2 17:16 ns
-r--r--r--.   1 root       root       0 Feb  2 17:16 numa_maps
-rw-r--r--.   1 root       root       0 Feb  2 17:16 oom_adj
-r--r--r--.   1 root       root       0 Feb  2 17:16 oom_score
-rw-r--r--.   1 root       root       0 Feb  2 17:16 oom_score_adj
-r--------.   1 root       root       0 Feb  2 17:16 pagemap
-r--------.   1 root       root       0 Feb  2 17:16 personality
-rw-r--r--.   1 root       root       0 Feb  2 17:16 projid_map
lrwxrwxrwx.   1 root       root       0 Feb  2 17:16 root -> /
-rw-r--r--.   1 root       root       0 Feb  2 17:16 sched
-r--r--r--.   1 root       root       0 Feb  2 17:16 schedstat
-r--r--r--.   1 root       root       0 Feb  2 16:39 sessionid
-rw-r--r--.   1 root       root       0 Feb  2 17:16 setgroups
-r--r--r--.   1 root       root       0 Feb  2 17:16 smaps
-r--r--r--.   1 root       root       0 Feb  2 17:16 smaps_rollup
-r--------.   1 root       root       0 Feb  2 17:16 stack
-r--r--r--.   1 root       root       0 Feb  2 16:58 stat
-r--r--r--.   1 root       root       0 Feb  2 17:16 statm
-r--r--r--.   1 root       root       0 Feb  2 16:36 status
-r--------.   1 root       root       0 Feb  2 17:16 syscall
dr-xr-xr-x.  16 minio-user minio-user 0 Feb  2 17:16 task
-r--r--r--.   1 root       root       0 Feb  2 17:16 timers
-rw-rw-rw-.   1 root       root       0 Feb  2 17:16 timerslack_ns
-rw-r--r--.   1 root       root       0 Feb  2 17:16 uid_map
-r--r--r--.   1 root       root       0 Feb  2 17:16 wchan

```

## Your Environment
Minio service file:

```
[Unit]
Description=MinIO
Documentation=https://docs.min.io
Wants=network-online.target
After=network-online.target
AssertFileIsExecutable=/usr/local/bin/minio

[Service]
WorkingDirectory=/usr/local/

User=minio-user
Group=minio-user
ProtectProc=invisible

EnvironmentFile=/etc/default/minio
ExecStartPre=/bin/bash -c "if [ -z \"${MINIO_VOLUMES}\" ]; then echo \"Variable MINIO_VOLUMES not set in /etc/default/minio\"; exit 1; fi"
ExecStart=/usr/local/bin/minio server $MINIO_OPTS $MINIO_VOLUMES

# Let systemd restart this service always
Restart=always

# Specifies the maximum file descriptor number that can be opened by this process
LimitNOFILE=1048576

# Specifies the maximum number of threads this process can create
TasksMax=infinity

# Disable timeout logic and wait until process is stopped
TimeoutStopSec=infinity
SendSIGKILL=no

[Install]
WantedBy=multi-user.target

# Built for ${project.name}-${project.version} (${project.name})
```

* Version used: RELEASE.2021-11-24T23-19-33Z
* Operating System: Linux 5.4.17

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
ignore unreadable metrics on certain closed systems

## Motivation and Context
fixes #14233

## How to test this PR?
As per #14233 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by poornas — APPROVED** [review]

File: cmd/metrics-v2.go [hunk 1]
```diff
@@ -942,94 +942,109 @@ func getMinioProcMetrics() *MetricsGroup {
 			logger.LogOnceIf(ctx, err, nodeMetricNamespace)
```
Comment (anjalshireesh) [inline cmd/metrics-v2.go:942]: I reproduced the error, and looks like it is getting triggered at this line (942).

Comment (harshavardhana) [reply cmd/metrics-v2.go:942]: This line of error is okay and we should print this error at least @anjalshireesh not the other errors.

Comment (harshavardhana) [reply cmd/metrics-v2.go:942]: that is why it is left here on purpose

**Review by anjalshireesh — APPROVED** [review]

**Pull request merged**